tests/cmdline/commands/test_setup.py

```diff
@@ -205,11 +205,12 @@ def test_setup_profile_uuid(self):
         postgres.create_db(db_user, db_name)
 
         profile_name = 'profile-copy'
+        user_email = 'some@email.com'
         profile_uuid = str(uuid.uuid4)
         options = [
             '--non-interactive', '--profile', profile_name, '--profile-uuid', profile_uuid, '--db-backend',
             self.storage_backend_name, '--db-name', db_name, '--db-username', db_user, '--db-password', db_pass,
-            '--db-port', self.pg_test.dsn['port']
+            '--db-port', self.pg_test.dsn['port'], '--email', user_email
```

Comment:
This test started failing with the changes introduced here. I am unsure if that means that I broke something, or whether the test worked by accident.

Reply:
This actually looks like a regression because you made the `default` for this option lazy with a functools partial. I don't think the partial is properly called, so the value that is passed for the option is the partial and not the actual config option value

Reply:
Okay, after going through the guts of click and its modification in aiida, I've determined that the default value seems to be evaluating correctly, i.e. the callback is called, but it returns an empty tuple. I think the test worked previously because the default was evaluated before the actual test, and hence the `get_config('autofill.user.email')` was called in the local context, not in test context, where this value seems to be unavailable.

@sphuber could you maybe try if you can make the test fail locally on main branch in a fresh aiida install without any profile? That would I think confirm my hypothesis.

Reply:
You are right. The problem is that the `cmd_setup` module is imported at the top of the test file. This will evaluate the `setup` command and with it the defaults. So even if I add the `empty_config` fixture to the `test_setup_profile_uuid` test, by the time that fixture gets invoked to provide an empty config for the test, the defaults of the setup command will already have been set. So by making the defaults lazily evaluated, this would indeed explain the observed behavior.